rasa-ext-slate/src/Rasa/Ext/Slate/Internal/Event.hs

```diff
@@ -1,16 +1,19 @@
 module Rasa.Ext.Slate.Internal.Event (terminalEvents) where
 
 import Rasa.Ext
-
 import Rasa.Ext.Slate.Internal.State
 
+import Control.Monad
+
 import qualified Graphics.Vty as V
 
 -- | Provides keypress events from the terminal, converted from Vty events.
 terminalEvents :: Action ()
 terminalEvents = do
     v <- getVty
-    eventProvider $ convertEvent <$> V.nextEvent v
+    asyncEventProvider $ getEvents v
+      where
+        getEvents v dispatch = forever $ V.nextEvent v >>= dispatch . convertEvent
```

Comment:
Here's an example of the new way to do things. `getEvents` takes a dispatch function which it calls with each new event. Then you call `asyncEventProvider` with `getEvents`